framework/lib/components/step-stack/step-stack.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { useToken } from '@chakra-ui/system'
+import { addIndex, map } from 'ramda'
+import { StepStackProps } from './types'
+import { getChildrenWithProps } from '~/lib/utils'
+import { HStack, Stack } from '../stack'
+import { Center } from '../center'
+import { Divider } from '../divider'
+import { Label } from '../typography'
+
+export const StepStack = ({
+  children,
+  spacing = '4',
+  rowHeight = '10',
+  ...rest
+}: StepStackProps) => {
+  const rows = getChildrenWithProps(children, {})
+  const parsedRowHeight = useToken('sizes', rowHeight)
+
+  return (
+    <Stack spacing={ spacing } position="relative" { ...rest }>
+      { addIndex(map)(
```

Comment:
`addIndex` should not be called in here, you are creating a new function on every render. It should be moved to utils as `mapIndexed`.

HOWEVER: I though we did not want to use ramda in northligth and keep dependencies to a minimum so I was a bit surprised to see it here. 
In react components we do accept `list.map(...)` since the callback function is never reused. 
Or do you prefer this style?